### PR TITLE
[docs-only] Document running tests using ociswrapper

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
Added documentation on running tests with ociswrapper for env config test features

- [x] readme for ociswrapper
- [x] add doc in testing section
- [x] update local testing docker-compose setup

## Related Issue
Follow up https://github.com/owncloud/ocis/pull/6062

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/ociswrapper/README.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
+## oCIS Wrapper
+
+A tool to wraps the oCIS binary and provides a way to re-configure the running oCIS instance.
```
Comment (saw-jan) [inline tests/ociswrapper/README.md:3]:
```suggestion
A tool to wrap the oCIS binary and provides a way to re-configure the running oCIS instance.
```

File: tests/ociswrapper/README.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+When run, **ociswrapper** starts an API server that exposes some endpoints to re-configure the oCIS server.
+
+### Usage
+
+1. Build
```
Comment (ScharfViktor) [inline tests/ociswrapper/README.md:9]: `[list-item-indent] Incorrect list-item indent: add 1 space`

File: tests/ociswrapper/README.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+   ```bash
+   make build
+   ```
+
+2. Run
+
+   ```bash
+   ./bin/ociswrapper serve --bin=<path-to-ocis-binary>
```
[the remark this answers is not in the document]
Comment (saw-jan) [reply tests/ociswrapper/README.md:18]: Yeah, we can change it to other than `5000`

[the remark this answers is not in the document]
Comment (saw-jan) [reply tests/ociswrapper/README.md:18]: Changed it to `5200`. WIll it be suitable?

File: docs/ocis/development/testing.md [hunk 4]
```diff
@@ -193,23 +230,105 @@ If you want to work on a specific issue
...
 
 2. the tests will fail, try to understand how and why they are failing
 3. fix the code
 4. go back to 1. and repeat till the tests are passing.
 5. remove those tests from the expected failures file
 6. make a PR that has the fixed code, and the relevant lines removed from the expected failures file.
 
+## Running ENV config tests (@env-config)
+
+Test suites tagged with `@env-config` are used to test the environment variables that are used to configure oCIS. These tests are special tests that require oCIS server to be run using [ociswrapper](https://github.com/owncloud/ocis/blob/master/tests/ociswrapper/README.md).
```
Comment (phil-davis) [inline docs/ocis/development/testing.md:252]:
```suggestion
Test suites tagged with `@env-config` are used to test the environment variables that are used to configure oCIS. These tests are special tests that require the oCIS server to be run using [ociswrapper](https://github.com/owncloud/ocis/blob/master/tests/ociswrapper/README.md).
```

File: docs/ocis/development/testing.md [hunk 5]
```diff
@@ -193,23 +230,105 @@ If you want to work on a specific issue
...
+TEST_WITH_GRAPH_API=true \
+TEST_OCIS=true \
+TEST_SERVER_URL="https://localhost:9200" \
+BEHAT_FEATURE=tests/acceptance/features/apiAsyncUpload/delayPostprocessing.feature \
+make test-acceptance-api
+```
+
+### Writing new ENV config tests
+
+While writing tests for a new oCIS ENV configuration, please make sure to follow the below guidelines:
```
Comment (phil-davis) [inline docs/ocis/development/testing.md:285]:
```suggestion
While writing tests for a new oCIS ENV configuration, please make sure to follow these guidelines:
```

File: docs/ocis/development/testing.md [hunk 6]
```diff
@@ -193,23 +230,105 @@ If you want to work on a specific issue
...
+
+1. Tag the test suite (or test scenarios) with `@env-config`
+2. Use `OcisConfigHelper.php` for helper functions - provides functions to reconfigure running oCIS instance.
```
Comment (phil-davis) [inline docs/ocis/development/testing.md:288]:
```suggestion
2. Use `OcisConfigHelper.php` for helper functions - provides functions to reconfigure the running oCIS instance.
```

File: docs/ocis/development/testing.md [hunk 7]
```diff
@@ -193,23 +230,105 @@ If you want to work on a specific issue
...
+3. Recommended: add the new step implementations in `OcisConfigContext.php`
+
+## Running test suite with email service (@email)
+
+Test suites that are tagged with `@email` require email service. We use inbucket as the email service in our tests.
```
Comment (phil-davis) [inline docs/ocis/development/testing.md:293]:
```suggestion
Test suites that are tagged with `@email` require an email service. We use inbucket as the email service in our tests.
```

File: tests/ociswrapper/README.md [hunk 8]
```diff
@@ -0,0 +1,54 @@
+## oCIS Wrapper
+
+A tool to wrap the oCIS binary and provides a way to re-configure the running oCIS instance.
```
Comment (phil-davis) [inline tests/ociswrapper/README.md:3]:
```suggestion
A tool that wraps the oCIS binary and provides a way to re-configure the running oCIS instance.
```

**Review by phil-davis — COMMENTED** [review]
Just some minor English edits. I haven't actually tried all the instructions.

**Review by individual-it — CHANGES_REQUESTED** [review]
1. changes that are not related to the ociswrapper should be in an other PR
2. `WITH_WRAPPER=true` should be the default setting, so that developers who run the tests can run any test without having to think about it (separate PR)

File: tests/ociswrapper/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ScharfViktor) [reply tests/ociswrapper/README.md:18]: Yes, port 5200 is free. Thanks

image

[the remark this answers is not in the document]
Comment (saw-jan) [reply tests/ociswrapper/README.md:18]: I have made some changes in this PR to make it easy to review.
So, I will change the port number in next PR

File: docs/ocis/development/testing.md [hunk 9]
```diff
@@ -193,23 +230,105 @@ If you want to work on a specific issue
...
+
+# run oCIS
+PROXY_ENABLE_BASIC_AUTH=true \
+./bin/ociswrapper serve --bin=../../ocis/bin/ocis
+```
+
+### Run the tests
+
+```bash
+OCIS_WRAPPER_URL=https://localhost:5000 \
```
[the remark this answers is not in the document]
Comment (saw-jan) [reply docs/ocis/development/testing.md:248]: Will be done in the next PR

File: docs/ocis/development/testing.md [hunk 10]
```diff
@@ -45,7 +45,17 @@ For example `make -C tests/acceptance/docker localApiTests-apiAccountsHashDiffic
...
+{{< hint info >}}
+While running the tests, oCIS server is started with [ociswrapper](https://github.com/owncloud/ocis/blob/master/tests/ociswrapper/README.md) by default. In order to run the tests without ociswrapper, provide `WITH_WRAPPER=false` when running the tests. For example:
+
+```bash
+WITH_WRAPPER=false \
+BEHAT_FEATURE='tests/acceptance/features/apiAccountsHashDifficulty/addUser.feature:21' \
+make -C tests/acceptance/docker test-ocis-feature-ocis-storage
+```
+
+But some test suites that are tagged `@env-config` require the oCIS server to be run with ociswrapper. So, running those tests require `WITH_WRAPPER=true`.
```
Comment (individual-it) [inline docs/ocis/development/testing.md:57]:
```suggestion
But test suites that are tagged with `@env-config` require the oCIS server to be run with ociswrapper. So, running those tests require `WITH_WRAPPER=true` (default setting).
```

File: tests/acceptance/docker/Makefile [hunk 11]
```diff
@@ -26,6 +26,10 @@ else
 endif
 OCIS_IMAGE_TAG ?= dev
 
+# runn tests with ociswrapper by default
```
Comment (individual-it) [inline tests/acceptance/docker/Makefile:29]:
```suggestion
# run tests with ociswrapper by default
```

**Review by individual-it — CHANGES_REQUESTED** [review]

**Review by individual-it — APPROVED** [review]

**Pull request merged**